src/Symfony/Bundle/FrameworkBundle/Command/ServerStartCommand.php

```diff
@@ -177,25 +182,48 @@ private function isOtherServerProcessRunning($address)
     }
 
     /**
+     * Determine the absolute file path for the router script based on user input and the environment.
+     *
+     * @param InputInterface  $input  An InputInterface instance
+     * @param OutputInterface $output An OutputInterface instance
+     * @param string          $env    The application environment
+     *
+     * @return string|bool The absolute file path of the router script, or false on failure
+     */
+    private function determineRouterScript(InputInterface $input, OutputInterface $output, $env)
+    {
+        $router = $input->getOption('router');
+
+        if (null === $router) {
+            $router = $this
+                ->getContainer()
+                ->get('kernel')
+                ->locateResource(sprintf('@FrameworkBundle/Resources/config/router_%s.php', $env))
+            ;
+        } else {
+            if (!file_exists($router)) {
+                $output->writeln(sprintf('<error>The given router script "%s" does not exist</error>', $router));
```

Comment:
Printing the error message here might not be enough. The method also returns `false` when `realpath()` returns `false`. In that case, no error message would be shown to the user while the exit code would still be 1.